lease/leasehttp/http.go

```diff
@@ -188,9 +188,14 @@ func TimeToLiveHTTP(ctx context.Context, id lease.LeaseID, keys bool, url string
 			errc <- err
 			return
 		}
-		b, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
+		b, err = readFunc(resp)
 		if err != nil {
+			// follower cannot reach leader at the moment
+			// possibly from network errors
+			// translate to ErrNoLeader, expecting clients to retry
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				err = rpctypes.ErrNoLeader
```

Comment:
I think we also need to convert the internal error to grpc error for lease in the API layer.
https://github.com/coreos/etcd/blob/master/etcdserver/api/v3rpc/lease.go#L57